### DISCUSSION TITLE
Thunderbird can login but can't send / recieve

### DISCUSSION THREAD

**Discussion by JamesM92** [docker-mailserver/docker-mailserver#2575] [Q&A] [answered]
Got my docker container up and running,

setup my nginx proxy manager to allow the ports 25/143/587

updates my dns record with the dkim keys

created the empty postfix-access.cf and postfix-recieve.cf

I'm able to connect to the server and login with thunderbird, 

But then i am unable to send or recieve emails.

It keeps saying  have the wrong password, but I'm sure I've entered it correctly, and any time i try the other password times, plain/encrypted etc, it says the server wants the plain password.

here are the logs of the container booting, me connecting with thunderbird, and then me attempting to send an email via thunder bird.
-14:56:17 start of thunderbird checking server
-14:56:36 confirm log-in
-14:56:55 end of logging in
-14:58:46 start of sending email

```


[   INF   ]  Welcome to docker-mailserver 11.0.0
[   INF   ]  Initializing setup
[   INF   ]  Checking configuration
[   INF   ]  Configuring mail server
Nameservers 127.0.0.11
[   INF   ]  Post-configuration checks
[   INF   ]  Starting miscellaneous tasks
[   INF   ]  Starting daemons
[   INF   ]  penguinarmy.club is up and running
May  4 14:54:52 penguinarmy amavis[759]: starting. /usr/sbin/amavisd-new at penguinarmy.club amavisd-new-2.11.1 (20181009), Unicode aware, LC_CTYPE="C.UTF-8"
May  4 14:54:52 penguinarmy amavis[759]: perl=5.032001, user=, EUID: 112 (112);  group=, EGID: 114 114 (114 114)
May  4 14:54:52 penguinarmy amavis[759]: Net::Server: Group Not Defined.  Defaulting to EGID '114 114'
May  4 14:54:52 penguinarmy amavis[759]: Net::Server: User Not Defined.  Defaulting to EUID '112'
May  4 14:54:52 penguinarmy amavis[759]: No ext program for   .zoo, tried: zoo
May  4 14:54:52 penguinarmy amavis[759]: No ext program for   .doc, tried: ripole
May  4 14:54:52 penguinarmy amavis[759]: No decoder for       .F   
May  4 14:54:52 penguinarmy amavis[759]: No decoder for       .doc 
May  4 14:54:52 penguinarmy amavis[759]: No decoder for       .zoo 
May  4 14:54:52 penguinarmy amavis[759]: Using primary internal av scanner code for ClamAV-clamd
May  4 14:54:52 penguinarmy amavis[759]: Found secondary av scanner ClamAV-clamscan at /usr/bin/clamscan
May  4 14:54:53 penguinarmy postfix/master[1935]: daemon started -- version 3.5.6, configuration /etc/postfix
May  4 14:56:17 penguinarmy dovecot: imap-login: Aborted login (no auth attempts in 0 secs): user=<>, rip=192.168.0.1, lip=172.19.0.7, session=<4wwlLjTeWsrAqAAB>
May  4 14:56:17 penguinarmy dovecot: imap-login: Aborted login (no auth attempts in 0 secs): user=<>, rip=192.168.0.1, lip=172.19.0.7, session=<hRMlLjTeW8rAqAAB>
May  4 14:56:17 penguinarmy dovecot: imap-login: Aborted login (no auth attempts in 0 secs): user=<>, rip=192.168.0.1, lip=172.19.0.7, session=<3aElLjTeXMrAqAAB>
May  4 14:56:17 penguinarmy dovecot: imap-login: Aborted login (no auth attempts in 0 secs): user=<>, rip=192.168.0.1, lip=172.19.0.7, session=<0SYmLjTeXsrAqAAB>
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: CONNECT from [192.168.0.1]:51821 to [172.19.0.7]:25
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: CONNECT from [192.168.0.1]:51822 to [172.19.0.7]:25
May  4 14:56:17 penguinarmy postfix/submission/smtpd[2170]: connect from unknown[192.168.0.1]
May  4 14:56:17 penguinarmy postfix/submission/smtpd[2173]: connect from unknown[192.168.0.1]
May  4 14:56:17 penguinarmy postfix/submission/smtpd[2170]: disconnect from unknown[192.168.0.1] ehlo=1 quit=1 commands=2
May  4 14:56:17 penguinarmy postfix/submission/smtpd[2173]: disconnect from unknown[192.168.0.1] ehlo=1 quit=1 commands=2
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: CONNECT from [192.168.0.1]:51828 to [172.19.0.7]:25
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: CONNECT from [192.168.0.1]:51831 to [172.19.0.7]:25
May  4 14:56:17 penguinarmy postfix/submission/smtpd[2170]: connect from unknown[192.168.0.1]
May  4 14:56:17 penguinarmy postfix/submission/smtpd[2170]: disconnect from unknown[192.168.0.1] ehlo=1 quit=1 commands=2
May  4 14:56:17 penguinarmy postfix/submission/smtpd[2173]: connect from unknown[192.168.0.1]
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: HANGUP after 0.18 from [192.168.0.1]:51828 in tests before SMTP handshake
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: DISCONNECT [192.168.0.1]:51828
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: HANGUP after 0.06 from [192.168.0.1]:51831 in tests before SMTP handshake
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: DISCONNECT [192.168.0.1]:51831
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: HANGUP after 0.35 from [192.168.0.1]:51821 in tests before SMTP handshake
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: DISCONNECT [192.168.0.1]:51821
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: HANGUP after 0.35 from [192.168.0.1]:51822 in tests before SMTP handshake
May  4 14:56:17 penguinarmy postfix/postscreen[2174]: DISCONNECT [192.168.0.1]:51822
May  4 14:56:17 penguinarmy postfix/submission/smtpd[2173]: lost connection after CONNECT from unknown[192.168.0.1]
May  4 14:56:17 penguinarmy postfix/submission/smtpd[2173]: disconnect from unknown[192.168.0.1] commands=0/0
May  4 14:56:36 penguinarmy dovecot: auth: passwd-file(admin,192.168.0.1,<GeJFLzTegcrAqAAB>): unknown user (SHA1 of given password: 32a22b)
May  4 14:56:42 penguinarmy dovecot: auth: passwd-file(admin,192.168.0.1,<GeJFLzTegcrAqAAB>): unknown user (SHA1 of given password: 32a22b)
May  4 14:56:48 penguinarmy dovecot: auth: passwd-file(admin,192.168.0.1,<GeJFLzTegcrAqAAB>): unknown user (SHA1 of given password: 32a22b)
May  4 14:56:50 penguinarmy dovecot: imap-login: Disconnected (auth failed, 3 attempts in 14 secs): user=<admin>, method=PLAIN, rip=192.168.0.1, lip=172.19.0.7, TLS, session=<GeJFLzTegcrAqAAB>
May  4 14:56:54 penguinarmy dovecot: imap-login: Login: user=<admin@penguinarmy.club>, method=PLAIN, rip=192.168.0.1, lip=172.19.0.7, mpid=2284, TLS, session=<xL4gMDTeicrAqAAB>
May  4 14:56:54 penguinarmy dovecot: imap(admin@penguinarmy.club)<2284><xL4gMDTeicrAqAAB>: Logged out in=9 out=480 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
May  4 14:56:55 penguinarmy dovecot: imap-login: Login: user=<admin@penguinarmy.club>, method=PLAIN, rip=192.168.0.1, lip=172.19.0.7, mpid=2286, TLS, session=<JoRjMDTej8rAqAAB>
May  4 14:58:46 penguinarmy postfix/submission/smtpd[2595]: connect from unknown[192.168.0.1]
May  4 14:58:46 penguinarmy postfix/submission/smtpd[2595]: Anonymous TLS connection established from unknown[192.168.0.1]: TLSv1.2 with cipher ECDHE-ECDSA-AES128-GCM-SHA256 (128/128 bits)
May  4 14:58:46 penguinarmy dovecot: auth: passwd-file(admin,192.168.0.1): unknown user (SHA1 of given password: 32a22b)
May  4 14:58:48 penguinarmy postfix/submission/smtpd[2595]: warning: unknown[192.168.0.1]: SASL PLAIN authentication failed: 
May  4 14:58:52 penguinarmy dovecot: auth: passwd-file(admin,192.168.0.1): unknown user (SHA1 of given password: 32a22b)
May  4 14:58:54 penguinarmy postfix/submission/smtpd[2595]: warning: unknown[192.168.0.1]: SASL LOGIN authentication failed: UGFzc3dvcmQ6


```

**Comment by casperklein** [comment]
> setup my nginx proxy manager to allow the ports 25/143/587

Afaik NPM only supports HTTP(S).

Edit: TIL this can be done with "streams" in NPM.

**Reply by JamesM92** [reply]
actually i did it with both http's and streams, each seems to allow thunderbird to connect.

But neither seems to be able to send the message,

Nor does using both resolve it.

**Comment by casperklein** [comment]
I would start with a basic setup and once that is working, start adding other things like NPM.

> auth: passwd-file(admin,192.168.0.1): unknown user

This looks like you did not provide a domain for the "admin" user? You probably want something like `admin@penguinarmy.club`.

**Comment by JamesM92** [comment]
Still been trying to get this to work off and on since may. 

I have done every combination of port, port stream, reverse proxy naming, mx record, spf record, I've tried using the basic compose and no matter what i do i get stuck with being able to login to the account, it seeing the server, and it hanging up when it tries to send an email. 


![image](https://user-images.githubusercontent.com/36548759/184945830-c9e7fdae-ef58-4083-9ee2-d58ec9891061.png)
![image](https://user-images.githubusercontent.com/36548759/184945866-1e170c7d-07ff-48c4-aa24-c51cd48ffe7a.png)
![image](https://user-images.githubusercontent.com/36548759/184945974-2bfb8095-c53b-4e13-bc39-beb8073d4f83.png)

i try and send an email to myself, it sits trying for a long time then give the error.


I've setup every port in question to be able to stream to the server bypassing the reverse proxy.
![image](https://user-images.githubusercontent.com/36548759/184946344-670fd4ad-ea33-40b3-9edb-f83cdcefbb11.png)

I've opened every port on my router
![image](https://user-images.githubusercontent.com/36548759/184946440-0870f4b0-f4d2-40a3-977f-3afb9370f70c.png)

i setup reverse proxies for the smtp and imap ports in case they were needed
![image](https://user-images.githubusercontent.com/36548759/184947175-275ee95b-5c65-4c51-a8bc-6d726b12826d.png)
![image](https://user-images.githubusercontent.com/36548759/184946565-d722b359-f212-4811-9f4f-4a5d7691d22b.png)
![image](https://user-images.githubusercontent.com/36548759/184946618-c798fb8d-afe5-45a5-bded-65601a10f45e.png)



I've created the dkim files and loaded the records to my DNS, for this im using go daddy.
![image](https://user-images.githubusercontent.com/36548759/184946872-7d6a4c97-899d-4815-bf2e-ffe95baa0b42.png)
![image](https://user-images.githubusercontent.com/36548759/184946989-808a6887-d50c-41d9-8338-f32d1c5e60e5.png)
![image](https://user-images.githubusercontent.com/36548759/184947021-32da959e-ca20-4d8b-bf0a-3d0ca6c6e6b3.png)



i have been trying to get this to work for months, and I have no clue what it is Im doing wrong.

**Reply by polarathene** [reply]
> I have done every combination of port, port stream, reverse proxy naming, mx record, spf record, I've tried using the basic compose and no matter what i do i get stuck with being able to login to the account, it seeing the server, and it hanging up when it tries to send an email.

First get this working locally, I can connect with Thunderbird to a local instance and all I had to do was expose the port and edit my `/etc/hosts` to point the hostname to localhost, then login with the full account (eg: `user@example.com`).

Once you get that working, then run it externally, then add in NPM if you want that.

---

> i setup reverse proxies for the smtp and imap ports in case they were needed

Those says they're for HTTP, none of those ports should be configured for that.

Also note that ports 25 and 587 speak StartTLS, any reverse proxy needs to handle that AFAIK. Port 465 should work (uses normal TLS).

**Reply by JamesM92** [reply]
I'm running thunderbird on a windows pc, and the mailserver on a headless linux server, I can't get thunderbird to accept an ip address or the computer name as the hostname to connect to.

also for reference here is the docker compose for the basic recommended install.
```
version: "3"
#43
networks:
  frontend:
    external: 
      name: FrontEnd
  backend:
    external:
      name: BackEnd

volumes:
  config:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /srv/dev-disk-by-label-Archive/portainer/MailServer/_data/config
    
  data:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /srv/dev-disk-by-label-Archive/portainer/MailServer/_data/data
 
  state:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /srv/dev-disk-by-label-Archive/portainer/MailServer/_data/state
 
  log:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /srv/dev-disk-by-label-Archive/portainer/MailServer/_data/logs
  
 
 
services:
  server:
    image: docker.io/mailserver/docker-mailserver:latest
    container_name: 03_MailServer
    restart: unless-stopped
    
    # If the FQDN for your mail-server is only two labels (eg: example.com),
    # you can assign this entirely to `hostname` and remove `domainname`.
    #hostname: mail
    #domainname: penguinarmy.club
    hostname: penguinarmy.club
    
    #env_file: stack.env
    environment:
      - ENABLE_SPAMASSASSIN=1
      - SPAMASSASSIN_SPAM_TO_INBOX=1
      - ENABLE_CLAMAV=1
      - ENABLE_FAIL2BAN=1
      - ENABLE_POSTGREY=1
      - ENABLE_SASLAUTHD=0
      - ONE_DIR=1
    volumes:
      - data:/var/mail/
      - state:/var/mail-state/
      - log:/var/log/mail/
      - config:/tmp/docker-mailserver/
      - /etc/localtime:/etc/localtime:ro
    #command:
    ports:
      - "25:25"    # SMTP  (explicit TLS => STARTTLS)
      - "143:143"  # IMAP4 (explicit TLS => STARTTLS)
      - "465:465"  # ESMTP (implicit TLS)
      - "587:587"  # ESMTP (explicit TLS => STARTTLS)
      - "993:993"  # IMAP4 (implicit TLS)
    cap_add:
      - NET_ADMIN
    networks:
      - frontend
      - backend
    stop_grace_period: 1m
    healthcheck:
      test: "ss --listening --tcp | grep -P 'LISTEN.+:smtp' || exit 1"
      timeout: 3s
      retries: 0
```

**Reply by polarathene** [reply]
I'm unfortunately stretched for time to assist you any further.

I can assure you that I recently had success getting Thunderbird to connect to a DMS container (_I just used `docker run`, no `docker-compose.yml`_) running all in a single VMware VM guest (Fedora 36). My container `hostname` would have been something like `example.test`, and all I did was add an entry into `/etc/hosts` to resolve that to `localhost` and Thunderbird could connect to the container to send/receive mail.

Perhaps you should start with that simpler setup before you broaden it to the real environment you're trying to get it working with. Start small and work upwards.

**Reply by JamesM92** [reply]
At this point I'd be willing to pay somebody $50 to help me get this working.

I'm telling it to connect to the server via the IP address but thunderbird complains the incoming port already exists. But I've made sure to remove all previous records from thunderbird.

I got the letsencrypt cert working and got thunderbird to recognize both ports as starttls, but it only did it once, then the imap stopped reporting it was starttls and I can't get it to do it again.

The few times it did work it complained the user didn't exist, but I've trippled checked that it was there.

**Comment by JamesM92** [comment, marked as answer]
Ok, I think i see whats causing the issue, but i don't know whats causing it.

In the server I've created the admin@penguinarmy.club account
![image](https://user-images.githubusercontent.com/36548759/185774223-702f3441-d295-48f7-88fb-2d8f938f9d9a.png)

in Thunderbird I'm entering the admin@penguinarmy.club 
![image](https://user-images.githubusercontent.com/36548759/185774237-d0af82b4-b1ef-4df5-a35d-5417d251bc92.png)

but when the server tries to process the login its only seeing the user name and not the full email.
![image](https://user-images.githubusercontent.com/36548759/185774265-66f1302f-7f46-471d-83dd-d8838ab11a26.png)

Thunderbird tries multiple attempts to login, after 3 fail2ban kicks in and bans the IP Im access the server from , which causes the imap connection to no longer report its starttls status.

this is because thunderbird by default changes the username its using for authentication to just the user part of the email address, unless you manually over ride it.

![image](https://user-images.githubusercontent.com/36548759/185774555-c18af915-8c06-45a6-9c13-b4f8e365104a.png)

I have no clue how many months  I have been scratching my head around this issue, but I'm glad to have finally solved it.

I've been able to successfully log in, and connect to the server with thunderbird now.

**Reply by polarathene** [reply]
Note, your `setup email list` implies you have made an alias for your account with the same address...why do you have that?

You can also try `ENABLE_QUOTAS=0` to disable the quota feature (_there is a workaround to support this feature regarding aliases_). But it still looks like a misconfiguration to create an alias address with the same address as a real account.

---

You can try reproduce with the following:

```bash
# Start a minimal container
docker run --rm -it --hostname example.test --name throwaway mailserver/docker-mailserver

# Open another shell/terminal to shell into the container, while you watch logs from the original terminal used prior:
docker exec -it throwaway bash

# Create an account
# NOTE: Since no existing accounts were configured, startup scripts have not completed while waiting for an account..
# There is a bug due to account creation not running after adding the first account manually, so another change needs
# to run to have the change detection trigger re-creation of the accounts (this is only a bug when no `postfix-accounts.cf` is mounted)
# https://github.com/docker-mailserver/docker-mailserver/blob/0b8b77ad886903824a84e161463c4ee27220e139/target/scripts/startup/setup-stack.sh#L275-L300
# Bugfix PR: https://github.com/docker-mailserver/docker-mailserver/pull/2731
setup email add hello@example.test pass
# wait a bit for setup scripts to complete, `supervisorctl status changedetector` should output `RUNNING`, then:
setup email add test@example.test pass
# This may take around 10 seconds for changedetector service to process and restart Dovecot, if everything is working then this command will work without errors:
setup email list

# Now you can check your account is authenticating correctly internally:
doveadm auth test hello@example.test pass
# If succesful you will see:
# "passdb: hello@example.test auth succeeded"
# Otherwise:
# "passdb: hello@example.test auth failed"
# Check the logs, failure should still show `passwd-file(hello@example.test` not `passwd-file(hello`
```

You can skip most of that to check with your existing container with the `doveadm auth test` command.

But if it's also outputting `admin` only, like in your log screenshot, then you should try the fresh basic container setup guide the rest of the commands provide.

If that then works correctly, create the container with some ports and connect to it from ThunderBird and verify that is working.

**Reply by polarathene** [reply]
Just saw your edit, glad you resolved it finally :)

**Reply by JamesM92** [reply]
thanks for all the assistance you've given.
